app/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+set(QT_NO_CREATE_VERSIONLESS_FUNCTIONS true)
+
+find_package(Qt6 6.3 COMPONENTS Quick REQUIRED)
+qt6_standard_project_setup()
+
+qt6_add_executable(${PROJECT_NAME} "")
+
+# TODO: Fix temporarly workaround until we make qt6_target_qml_sources work
+# Adds qml files as /Status/Application/qml/.../<file name>
+qt6_add_qml_module(${PROJECT_NAME}
+    URI Status.Application
+    VERSION 1.0
+    QML_FILES
+        qml/main.qml
+        qml/Status/Application/StatusWindow.qml
+        qml/Status/Application/StatusContentView.qml
+)
+
+# Compile time definitions required by the project
+target_compile_definitions(${PROJECT_NAME} PRIVATE BUILD_PROJECT_NAME=${PROJECT_NAME})
+target_compile_definitions(${PROJECT_NAME} PRIVATE BUILD_PROJECT_DOMAIN=${PROJECT_DOMAIN})
+target_compile_definitions(${PROJECT_NAME} PRIVATE BUILD_PROJECT_ORGANIZATION_NAME=${PROJECT_ORGANIZATION_NAME})
+target_compile_definitions(${PROJECT_NAME} PRIVATE BUILD_PROJECT_APPLICATION_NAME=${PROJECT_APPLICATION_NAME})
+
+target_compile_definitions(${PROJECT_NAME} PRIVATE BUILD_BINARY_DIR=${CMAKE_BINARY_DIR})
+target_compile_definitions(${PROJECT_NAME} PRIVATE BUILD_SOURCE_DIR=${CMAKE_SOURCE_DIR})
+
+add_subdirectory(qml)
+add_subdirectory(src)
+
+set_target_properties(${PROJECT_NAME} PROPERTIES
+    MACOSX_BUNDLE_GUI_IDENTIFIER ${PROJECT_NAME}.${STATUS_DOMAIN}
```

Comment:
Platform-specific stuff can be moved to separated files, and here you can just call `configure_os_specific()` what can be implimented on every platfom in different way

Reply:
Great recommendation I will follow in a later refactoring.